experiments/TestCase/baroclinic_wave_fvm.jl

```diff
@@ -0,0 +1,346 @@
+#!/usr/bin/env julia --project
+
+using ArgParse
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using ClimateMachine
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Thermodynamics:
+    air_density, air_temperature, total_energy, internal_energy, PhasePartition
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.VariableTemplates
+import ClimateMachine.DGMethods.FVReconstructions: FVLinear
+
+using CLIMAParameters
+using CLIMAParameters.Planet: MSLP, R_d, day, grav, Omega, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function init_baroclinic_wave!(problem, bl, state, aux, localgeo, t)
+    FT = eltype(state)
+
+    # parameters
+    _grav::FT = grav(bl.param_set)
+    _R_d::FT = R_d(bl.param_set)
+    _Ω::FT = Omega(bl.param_set)
+    _a::FT = planet_radius(bl.param_set)
+    _p_0::FT = MSLP(bl.param_set)
+
+    k::FT = 3
+    T_E::FT = 310
+    T_P::FT = 240
+    T_0::FT = 0.5 * (T_E + T_P)
+    Γ::FT = 0.005
+    A::FT = 1 / Γ
+    B::FT = (T_0 - T_P) / T_0 / T_P
+    C::FT = 0.5 * (k + 2) * (T_E - T_P) / T_E / T_P
+    b::FT = 2
+    H::FT = _R_d * T_0 / _grav
+    z_t::FT = 15e3
+    λ_c::FT = π / 9
+    φ_c::FT = 2 * π / 9
+    d_0::FT = _a / 6
+    V_p::FT = 1
+    M_v::FT = 0.608
+    p_w::FT = 34e3                 # Pressure width parameter for specific humidity
+    η_crit::FT = p_w / _p_0        # Critical pressure coordinate
+    q_0::FT = 0.018                # Maximum specific humidity (default: 0.018)
+    q_t::FT = 1e-12                # Specific humidity above artificial tropopause
+    φ_w::FT = 2π / 9               # Specific humidity latitude wind parameter
```

Comment:
I can't find if this is done for the other Experiments, too, but it would be great to document the choice of parameters and test case chosen, by perhaps leaving a reference?